source/addonHandler/__init__.py

```diff
@@ -517,26 +515,14 @@ def __init__(self, path: str):
 				_report_manifest_errors(self.manifest)
 				raise AddonError("Manifest file has errors.")
 
-	def completeInstall(self) -> Optional[str]:
-		if not os.path.exists(self.pendingInstallPath):
-			log.error(f"Pending install path {self.pendingInstallPath} does not exist")
-			return None
-
+	def completeInstall(self) -> str:

```

Comment:
This still returns None on error

```suggestion
	def completeInstall(self) -> Optional[str]:
```